http-client/src/main/java/io/airlift/http/client/spnego/SpnegoAuthentication.java

```diff
@@ -193,6 +161,58 @@ public boolean matches(String type, URI uri, String realm)
         return NEGOTIATE.equalsIgnoreCase(type);
     }
 
+    private void ensureClientCredentialUpdated()
+            throws LoginException, GSSException
+    {
+        if (clientCredential == null || clientCredential.getRemainingLifetime() <= MIN_CREDENTIAL_LIFE_TIME_SEC) {
+            if (loginContext != null) {
+                loginContext.logout();
```

Comment:
Why does it need to log out? Wouldn't it be sufficient to get a new client credential by calling `GSS_MANAGER.createCredential(...)`? I.e., is the login session or the client credential that expires?